### PR TITLE
improve ondemand test workflow

### PR DESCRIPTION
**Pull request by anirudhwarrier**
Added boolean inputs to automation-ondemand-tests to enable/disable run of chaos and reorg tests, since they are not needed to be run at all times.

### CODE REVIEW

File: .github/workflows/automation-ondemand-tests.yml [hunk 1]
```diff
@@ -104,6 +114,7 @@ jobs:
         with:
           ref: ${{ github.head_ref || github.ref_name }}
       - name: Build Test Image
+        if: inputs.enableChaos || inputs.enableReorg
```
Comment (infiloop2) [inline .github/workflows/automation-ondemand-tests.yml:117]: does this mean that the test image will be created only when chaos or reorg are enabled? Is that intended?

File: .github/workflows/automation-ondemand-tests.yml [hunk 2]
```diff
@@ -185,7 +204,7 @@ jobs:
             echo "upgrade_version=${{ inputs.chainlinkVersion }}" >>$GITHUB_OUTPUT
             echo "upgrade_image=$READ_CL_IMAGE" >>$GITHUB_OUTPUT
           fi
-          if [[ "${{ matrix.tests.name }}" == "upgrade" ]]; then
+          if [[ "${{ matrix.tests.type }}" == "upgrade" ]]; then
```
Comment (infiloop2) [inline .github/workflows/automation-ondemand-tests.yml:207]: is this unrelated change?

File: .github/workflows/automation-ondemand-tests.yml (hunk 1 above)
Comment (anirudhwarrier) [reply .github/workflows/automation-ondemand-tests.yml:117]: yes that is right. the test image is needed only for running chaos and reorg test which are tests on k8s. The upgrade tests run on docker within the github runner, so no need of test image.

File: .github/workflows/automation-ondemand-tests.yml (hunk 2 above)
Comment (anirudhwarrier) [reply .github/workflows/automation-ondemand-tests.yml:207]: this is a required change to fix the upgrade test.
